tests/drivers/flash/api/page_layout/src/main.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2020-2022 Nordic Semiconductor ASA
+ * Copyright (c) 2024 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/flash.h>
+
+#define TEST_DEVICE flash_sim0
```

Comment:
Use a "reserved" node alias here, and rely on board overlays to correctly label the flash device
```
#define TEST_DEVICE_ALIAS dut
#define TEST_NODE DT_ALIAS(TEST_DEVICE_ALIAS)
```
then, adding support for a board is done by adding a board overlay to `boards/`
```
/* Overlay for foo_board */
/{
        aliases {
                dut = &flash0;
        };
};
```

This allows for defining both the device to test, and potential board specific Kconfigs in the `.conf` file in `boards/`, without changing the tests source code :)